app/models/precompute_searchbar_json.rb

```diff
@@ -0,0 +1,33 @@
+class PrecomputeSearchbarJson
+
+  def initialize
+    @json_for_all_students = nil
+  end
+
+  def for_all_educators
+    Educator.find_each { |educator| self.for(educator) }
+  end
+
+  def for_educators_who_log_in
+    educators_who_log_in.find_each { |educator| self.for(educator) }
+  end
+
+  def for(educator)
+    if educator.districtwide_access
```

Comment:
@alexsoble @jhilde I'm concerned this approach in general, and about moving authorization code into `PrecomputeSearchbarJson` in particular.

If the purpose of this class is to help precompute some data that's slow, then it shouldn't be reaching into educator models to read their roles and determine who's authorized to see students.  If the problem is that it's expensive to compute authorization, can we look at speeding that up?  I submitted https://github.com/studentinsights/studentinsights/pull/1129 as a step towards that.